enhancements/declarative-index-config.md

```diff
@@ -487,20 +512,26 @@ Since each bundle in the index belongs to a channel in a package, the index itse
 
 The config file for each package will have a stream of json objects, representing the package and bundle information for that package. Currently, this information is stored in a sql database inside the index. To capture all the information previously stored in this database but in a normalized way, the config file will have two kinds of json blob: 
 
-1. The json blob capturing package information. 
+##### Declarative Package Format
 ```json
 {
     "schema": "olm.package",
     "name": "<package-name>",
     "defaultChannel": "<channel-name>",
     "icon": "embedded or remote",
-    "channels": ["<list-of-channels-in-package>"],
+    "validChannelNames": ["<optional-list-of-channels-in-package>"],
     "description": "A message about the operator, a description of channels..."
 }
 ```
+
+If `validChannelNames` is non-empty, it will be used to validate channel entries defined by bundles.
+This can help index maintainers ensure that bundles do not accidentally mis-name their channels,
+which could lead to the inadvertent creation of new channels
+(e.g. adding a new `stable-v1.0` channel to a bundle when the intention is to add it to the existing `stable-1.0` channel).
+
```

Comment:
I don't think we want to pollute the `olm.package` schema with validator parameters.  It's information that most clients of that schema don't need or care about.

Instead, I think we want to include these parameters in a way that makes them opaque to all clients other than the ones use them. Fortunately, because the package config spec allows for concatenation with arbitrary top-level JSON objects, it's possible for clients to define new object types for things like "validator parameters" without affecting preexisting schemas.

e.g.

```json
{
    "schema": "my-validator.channels",
    "package": "etcd",
    "allowed": ["singlenamespace-alpha", "clusterwide-alpha"]
}
{
    "schema": "olm.package",
    "name": "<package-name>",
    "defaultChannel": "<channel-name>",
    "icon": "embedded or remote",
    "description": "A message about the operator"
}
```

cc @bparees

Reply:
i like it

Reply:
That's fair. I'll remove the `channels` field from `olm.package` and we can cover this kind of thing separately.

Reply:
Done.